File: codes/OOP/cpp/encapsulation.cpp
//Data abstraction refers to, providing only essential information to the outside world and hiding their background details

//Encapsulation is an Object Oriented Programming concept that binds together the data and functions that manipulate the data,
//and that keeps both safe from outside interference and misuse.
#include<bits/stdc++.h>
using namespace std;

class Shape {
    int width, height;
public:
    int getWidth() {
        return width;
    }
    int getHeight() {
        return height;
    }
    void setWidth(int w) {
        width = w;
    }
    void setHeight(int h) {
        height = h;
    }
};

int main() {
    Shape s;
    s.setHeight(4);
    s.setWidth(6);
    cout << s.getHeight() << " " << s.getWidth();
    return 0;
}

File: codes/OOP/java/Encapsulation.java
/**
 * Created by gaurav on 6/2/16.
 */
class Shape {
    private int height, width;

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }
}

public class Encapsulation {
    public static void main(String[] args) {
        Shape shape = new Shape();
        shape.setHeight(5);
        shape.setWidth(4);
        System.out.println(shape.getHeight());
        System.out.println(shape.getWidth());
    }
}
